cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -1989,3 +2012,78 @@ def keystone_config():
             create_self_config(ca, client)
         generate_keystone_configmap()
         set_state('keystone.credentials.configured')
+
+
+@when('vault.available',
+      'layer.vaultlocker.ready')
+@when_not('kubernetes-master.secure-storage.created')
+def create_secure_storage():
+    encryption_conf_dir = encryption_config_path().parent
+    encryption_conf_dir.mkdir(mode=0o700, parents=True, exist_ok=True)
+    try:
+        device_name = unitdata.kv().get('kubernetes-master.loop-device')
+        if not device_name:
+            check_call(['modprobe', 'loop'])
+            crypt_file = '/root/cdk/encrypted-config'
+            check_call(['dd', 'if=/dev/urandom', 'of={}'.format(crypt_file),
+                        'bs=1M', 'count=20'])
+            check_call(['losetup', '-f', crypt_file])
+            losetup_output = check_output(['losetup', '--json'])
+            device_name = [d['name']
+                           for d in json.loads(losetup_output)['loopdevices']
+                           if d['back-file'] == crypt_file][0]
+            unitdata.kv().set('kubernetes-master.loop-device', device_name)
+        vaultlocker.encrypt_device(device_name, str(encryption_conf_dir))
+        set_flag('kubernetes-master.secure-storage.created')
```

Comment:
Should we also clear `kubernetes-master.secure-storage.failed` here?

Reply:
Yes.

Reply:
Addressed